File: PA2/04/lcm.py
# Uses python3
import sys

def lcm_naive(a, b):
    for l in range(1, a*b + 1):
        if l % a == 0 and l % b == 0:
            return l
    return a*b

def lcd_fast (dividend, divisor):   # JDH added
    if (divisor == 0):
        return dividend
    remainder = dividend % divisor
    return lcd_fast(divisor, remainder)

def lcm_fast (a, b):    # JDH added
    result = (a * b // lcd_fast (a, b)) # JDH use // so it's not a float
    return result

if __name__ == '__main__':
    input = sys.stdin.readline()
    a, b = map(int, input.split())
    # print(lcm_naive(a, b)) # JDH original
    print(lcm_fast(a, b))   # JDH added

File: PA4/lcm.cpp
#include <iostream>

long long lcm_naive(int a, int b) {
  for (long l = 1; l <= (long long) a * b; ++l)
    if (l % a == 0 && l % b == 0)
      return l;

  return (long long) a * b;
}

long lcd_fast (long dividend, long divisor) {	// JDH added
	long remainder = 0;

	if (divisor == 0)	// was <= 1
		return dividend;

	remainder = dividend % divisor;
	return (lcd_fast (divisor, remainder));
}

long long lcm_fast (long a, long b) {	// JDH added

	return (a * b / lcd_fast (a, b));
}

int main() {
  long a, b;	// JDH original int
  std::cin >> a >> b;
  // std::cout << lcm_naive(a, b) << std::endl;
  std::cout << lcm_fast(a, b) << std::endl;
  return 0;
}
